src/formatDate.js

```diff
@@ -9,6 +41,31 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+
+  const oldDateParts = date.split(oldSeparator);
+
+  let datePartsMatched = {};
+
+  for (let i = 0; i < oldDateParts.length; i++) {
+    datePartsMatched[fromFormat[i]] = oldDateParts[i];
+  }
+
+  datePartsMatched = convertYear(datePartsMatched, fromFormat, toFormat);
+
+  const newDateParts = [];
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    const newDateFormat = toFormat[i];
+
+    const newDatePart = datePartsMatched[newDateFormat];
+
+    newDateParts.push(newDatePart);
```

Comment:
Similar to the previous comment about `convertYear`, the code assumes that the `toFormat` array will only contain 'YYYY', 'YY', 'MM', and 'DD'. If the function should be flexible to handle other date parts in the future, it would be beneficial to handle this dynamically.